compiler/qsc_frontend/src/typeck/infer.rs

```diff
@@ -369,11 +399,36 @@ impl Inferrer {
                 self.constraints.push_front(constraint);
```

Comment:
I can't comment on the exact line as it's too far outside the diff, but the comment pointing to issue 152 above here can be removed with this fix.